mnemosyne/core/hygiene.py

```
@@ -227,107 +227,239 @@ def _scan_table(
     """Scan a table for noise candidates. Returns rows as dicts."""
     cursor = conn.cursor()
     # We deliberately select all columns we need; the schemas for
     # working_memory, memories, and episodic_memory all share the core
     # (id, content, source, timestamp, session_id, importance, metadata_json)
     # shape, with episodic_memory having extra columns we don't need.
-    cursor.execute(
+    base_query = (
         f"SELECT id, content, source, timestamp, session_id, importance, metadata_json "
-        f"FROM {table_name} ORDER BY timestamp DESC LIMIT ? OFFSET ?",
-        (limit, offset),
+        f"FROM {table_name}"
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)